test/unit/models/users.test.js

```diff
@@ -87,5 +87,62 @@ describe("users", function () {
       expect(user.last_name).to.equal(userData.last_name);
       expect(userExists).to.equal(true);
     });
+
+    it("should add the github_user_id to the user collection", async function () {
+      const userData = userDataArray[0];
+      userData.github_user_id = "123456789";
+
+      const { isNewUser, userId } = await users.addOrUpdate(userData);
+
+      const data = (await userModel.doc(userId).get()).data();
+
+      expect(data.github_user_id).to.equal(userData.github_user_id);
+      expect(isNewUser).to.equal(true);
+    });
+
+    it("should update the github_user_id in the user collection", async function () {
+      const userData = userDataArray[0];
+      userData.github_user_id = "123456789";
+
+      // Add the user the first time
+      const { userId } = await users.addOrUpdate(userData);
+
+      // Update the user with same data and new github_user_id
+      userData.github_user_id = "987654321";
+      await users.addOrUpdate(userData, userId);
+
+      const data = (await userModel.doc(userId).get()).data();
+
+      expect(data.github_user_id).to.equal(userData.github_user_id);
+    });
+
+    it("should be a string", async function () {
+      const userData = { ...userDataArray[0], github_id: 123 };
+
+      try {
+        await users.addOrUpdate(userData);
+      } catch (error) {
+        expect(error.message).to.equal("Validation error: github_id must be a string");
+      }
+    });
+
+    it("should have a maximum length of 9 characters", async function () {
+      const userData = { ...userDataArray[0], github_user_id: "a".repeat(9) };
+
+      try {
+        await users.addOrUpdate(userData);
+      } catch (error) {
+        expect(error.message).to.equal("Validation error: github_id exceeds maximum length of 50 characters");
+      }
+    });
```

Comment:
the test states max length 9 but here the error message says 50 !

Reply:
I'll change it to 9, thanks for figuring this out.

Reply:
how did we come to conclusion that 9 is the max length?

Reply:
Sorry I just counted again, max length should be 8 because the "GitHub_Id" that we fetch from api has a fixed lenght of 8. @RitikJaiswal75 
![Screenshot from 2023-05-02 11-10-50](https://user-images.githubusercontent.com/71075101/235587883-a49c4262-1884-4eb2-9db6-dbd1751019d1.png)